writers_rollingfilewriter.go

```diff
@@ -138,7 +140,7 @@ var compressionTypes = map[rollingArchiveType]compressionType{
 			if err != nil {
 				return err
 			}
-			return createGzip(archiveName, tar)
+			return createGzip(archiveName, bytes.NewReader(tar))
```

Comment:
I agree that we currently only call unarchive before archive, but I think the primitives are fine for two reasons:
1) we may want to call unarchive without calling archive on that type in the future: e.g. archive type conversion
2) we only call unarchive in one place in our repo

One use of `io.Pipe` for `unarchive`, `archive` and one use for `createTar`, `createGzip` is okay. It's when we chain several `io.Pipe`s together that I start questioning the overall architecture. Thoughts?